staging/src/k8s.io/utils/featuregates/composition_test.go

```diff
@@ -0,0 +1,34 @@
+package featuregates
+
+import "testing"
+
+func TestAddingSameFeatureGateTwice(t *testing.T) {
```

Comment:
test for basic composition:
1. can add the same featuregate multiple times transitively
2. cannot add duplicate of featuregates (not same instance)